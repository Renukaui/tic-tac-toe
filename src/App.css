.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 40vmin;
  pointer-events: none;
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

body {
  font: 14px "Century Gothic", Futura, sans-serif;
  margin: 0;
}

ol, ul {
  padding-left: 30px;
}

.board-row:after {
  clear: both;
  content: "";
  display: table;
}

.square {
  background: #fff;
  border: 1px solid #999;
  float: left;
  font-size: 4px;
  font-weight: bold;
  line-height: 34px;
  height: 34px;
  margin-right: -1px;
  margin-top: -1px;
  padding: 0;
  text-align: center;
  width: 34px;
}

.square:focus {
  outline: none;
}

.kbd-navigation .square:focus {
  background: #ddd;
}

/*
 * --- spicing up components ---
 */
/* square: 
 - bigger size and font
 - nice background color & transition
 - pointer cursor
*/
.square {
  font-size: 4rem;
  height: 140px;
  width: 140px;
  color: #cfc8b8;
  border: 4px solid #cfc8b8;
}
.square:hover {
  -webkit-transition: background-color 0.5s ease;
  transition: background-color 0.5s ease;
  background-color: #e6e2da;
  cursor: pointer;
}
.square.x {
  background-color: #524a4c;
}
.square.o {
  background-color: #6b948c;
}

.board {
  position: relative;
  border: 15px solid #cfc8b8;
  -webkit-box-shadow: 0px 0px 25px 0px rgba(0, 0, 0, 0.75);
  box-shadow: 0px 0px 25px 0px rgba(0, 0, 0, 0.75);
  z-index: 10;
}

h3 {
  text-align: center;
}
