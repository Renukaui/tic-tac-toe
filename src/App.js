import React, { Component } from 'react';
import Board from "./components/Board/Board";
import Moves from "./components/Moves/Moves";
import Status from "./components/Status/Status";
import calculateWinner from "./UtilityFunctions/calculateWinner";
import './App.css';

class Game extends Component {
	constructor(props){
		super(props);
		this.state = {
			history: [{ 
				squares:Array(9).fill(null), 
				clickIndex:null  
			}],
			xIsNext: true,
      stepNumber: 0,
      isTie: false
		};
	}

	handleClick = (i) => {
		const history = this.state.history.slice(0, this.state.stepNumber + 1);
		const current = history[history.length - 1];
		const squares = current.squares.slice();
		if(squares[i] || calculateWinner(squares)) {
			return;
    }
    squares[i] = this.state.xIsNext ? 'x':'o';
    const checkGameEnd = squares.filter(i => i === null);
    if (!checkGameEnd.length) {
      this.setState({
        history: history.concat([{ squares:squares, clickIndex:i }]),
            stepNumber	: history.length,
        isTie: !this.state.isTie
      })
    } else {
      this.setState({
        history: history.concat([{ squares:squares, clickIndex:i }]),
            stepNumber	: history.length,
        xIsNext: !this.state.xIsNext,
        isTie: this.state.isTie
      });
    }
	}

	jumpTo = (step) => {
    this.setState({
      stepNumber: step,
      xIsNext: (step % 2) === 0,
      isTie: !this.state.isTie
    });
	}
	
	render() {
		const history = this.state.history;
		const current = history[this.state.stepNumber];
		const squares = current.squares.slice();

		return(
			<div className="game">
				<div className="game-board">
					<Board squares={squares} onClick={this.handleClick} cells={[0,1,2,3,4,5,6,7,8]} />
				</div>
				<div className="game-info">
					<Status squares={ squares } xIsNext={ this.state.xIsNext } isTie={this.state.isTie}/>
					<Moves history={ this.state.history } stepNumber={ this.state.stepNumber } onClick={this.jumpTo} /> 
				</div>
			</div>
		);
	}
}

export default Game;
