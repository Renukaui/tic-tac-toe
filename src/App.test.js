import React from 'react';
import App from './App';
import Adapter from 'enzyme-adapter-react-16';
import {shallow, mount, configure} from 'enzyme';

configure({adapter: new Adapter()});

describe("Game", () => {
  let props;
  beforeEach(() => {
    props = {
      history: [{ 
          squares:Array(9).fill(null), 
          clickIndex:null  
      }]
    }
  })

  it('should match snapshot', () => {
    const wrapper = shallow(<App {...props}/>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render without crashing', () => {
    shallow(<App {...props}/>);
  });
  it('should play game and decide winner', () => {
    const wrapper = mount(<App {...props}/>);
    const firstPlayer = wrapper.find('.game-info__status').children().first().text()
    expect(firstPlayer).toEqual('Next player is: x');
    const button = wrapper.find('button.square').first();
    button.simulate('click');
    const secondPlayer = wrapper.find('.game-info__status').children().first().text();
    expect(secondPlayer).toEqual('Next player is: o');
    // player 2
    const turn2 = wrapper.find('button.square').at(1);
    turn2.simulate('click');
    // player 1
    const turn3 = wrapper.find('button.square').at(4);
    turn3.simulate('click');
    // player 2
    const turn4 = wrapper.find('button.square').at(5);
    turn4.simulate('click');
    // player 1
    const turn5 = wrapper.find('button.square').at(8);
    turn5.simulate('click');
    const winner = wrapper.find('div.game-info').children().first().text();
    expect(winner).toEqual('Winner is: x');
  })
});