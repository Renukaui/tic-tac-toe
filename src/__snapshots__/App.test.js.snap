// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should match snapshot 1`] = `ShallowWrapper {}`;

exports[`Game should match snapshot 1`] = `ShallowWrapper {}`;
